### DISCUSSION TITLE
Isothermal wall boundary condition

### DISCUSSION THREAD

**Discussion by kohyun** [su2code/SU2#1746] [Q&A]
Hi all,

I'm trying to simulate SWBLI VKI blunt body.
It requires a constant wall temperature boundary condition (Tw=300K).
I applied the following MARKER.

MARKER_ISOTHERMAL = (wall, 300.0)

But the result shows non-uniform temperature on the wall. (see figure below)

![image](https://user-images.githubusercontent.com/25341164/187032979-0b20f76f-2d06-4b71-bbc7-39609870f7f5.png)

For the adiabatic wall BC case, the simulation looks good but has some discrepancy from the experiment.

The followings are environments I am using. 

**Desktop (please complete the following information):**
 - OS: [e.g., Linux (Ubuntu 20.04)]
 - C++ compiler and version: [e.g., g++ (GCC) 9.4]
 - MPI implementation and version: [e.g., OpenMPI 4.03]
 - SU2 Version: [e.g., v7.4.0]

**Comment by WallyMaier** [comment]
Can you post the config file/conditions you are using?

**Reply by kohyun** [reply]
I have used the config file as following.

1. I have configured 2D axisymmetric case today to check further.
2. I had same non-uniform temperature distribution along the wall
![image](https://user-images.githubusercontent.com/25341164/187219345-8c6b97e0-dd31-4976-836b-28bee40d7a33.png)
![image](https://user-images.githubusercontent.com/25341164/187220348-ec913c6e-cab2-4d47-837f-6d1cd99a40a1.png)


============================================================================
=== config file for 2D axisymmetric case
============================================================================

SOLVER= NAVIER_STOKES
%
% Specify turbulence model (NONE, SA, SA_NEG, SST, SA_E, SA_COMP, SA_E_COMP, SST_SUST)
%KIND_TURB_MODEL= SA
%SA_OPTIONS=EDWARDS,COMPRESSIBILITY,EXPERIMENTAL
%

MATH_PROBLEM= DIRECT

AXISYMMETRIC=YES
%
%
% Restart solution (NO, YES)
RESTART_SOL= YES
%
SYSTEM_MEASUREMENTS= SI
%
%
% ------------------------------- SOLVER CONTROL ------------------------------%
%
% Maximum number of inner iterations
INNER_ITER= 50000
%
% Maximum number of outer iterations (only for multizone problems)
OUTER_ITER= 1
%
% Maximum number of time iterations
TIME_ITER= 1
%
% Convergence field 
CONV_FIELD= DRAG
%
% Min value of the residual (log10 of the residual)
CONV_RESIDUAL_MINVAL= -8
%
% Start convergence criteria at iteration number
CONV_STARTITER= 10
%
% Number of elements to apply the criteria
CONV_CAUCHY_ELEMS= 100
%
% Epsilon to control the series convergence
CONV_CAUCHY_EPS= 1E-10
%
% Iteration number to begin unsteady restarts
RESTART_ITER= 0
%


% -------------------- COMPRESSIBLE FREE-STREAM DEFINITION --------------------%
%
% Mach number (non-dimensional, based on the free-stream values)
MACH_NUMBER= 6.0
%
% Angle of attack (degrees, only for compressible flows)
AOA= 0.0
%
% Side-slip angle (degrees, only for compressible flows)
SIDESLIP_ANGLE= 0.0
%
% Init option to choose between Reynolds (default) or thermodynamics quantities
% for initializing the solution (REYNOLDS, TD_CONDITIONS)
INIT_OPTION= TD_CONDITIONS
%
% Free-stream option to choose between density and temperature (default) for
% initializing the solution (TEMPERATURE_FS, DENSITY_FS)
FREESTREAM_OPTION= TEMPERATURE_FS
%
% Free-stream pressure (101325.0 N/m^2, 2116.216 psf by default)
FREESTREAM_PRESSURE= 673.642
%
% Free-stream temperature (288.15 K, 518.67 R by default)
FREESTREAM_TEMPERATURE= 67.073
%
% Reynolds number (non-dimensional, based on the free-stream values)
REYNOLDS_NUMBER= 7.8E6
%
% Reynolds length (1 m, 1 inch by default)
REYNOLDS_LENGTH= 1.0
%
% Free-stream Turbulence Intensity (Not %, enter actual value, 0.001 = 0.1%)
% default value is 0.05 : 5%
% FLUENT default = 1%
%FREESTREAM_TURBULENCEINTENSITY = 0.001
%
% Free-stream Turbulent to Laminar viscosity ratio (defalut = 10.0)
FREESTREAM_TURB2LAMVISCRATIO = 10.0
%
% Compressible flow non-dimensionalization (DIMENSIONAL, FREESTREAM_PRESS_EQ_ONE,
%                              FREESTREAM_VEL_EQ_MACH, FREESTREAM_VEL_EQ_ONE)
REF_DIMENSIONALIZATION= DIMENSIONAL


% ---------------------- REFERENCE VALUE DEFINITION ---------------------------%
%
% Reference origin for moment computation (m or in)
REF_ORIGIN_MOMENT_X = 0.00
REF_ORIGIN_MOMENT_Y = 0.00
REF_ORIGIN_MOMENT_Z = 0.00
%
% Reference length for moment non-dimensional coefficients (m or in)
REF_LENGTH= 1.0
%
% Reference area for non-dimensional force coefficients (0 implies automatic
% calculation) (m^2 or in^2)
REF_AREA= 1.0
%

% ---- IDEAL GAS, POLYTROPIC, VAN DER WAALS AND PENG ROBINSON CONSTANTS -------%
%
% Fluid model (STANDARD_AIR, IDEAL_GAS, VW_GAS, PR_GAS,
%              CONSTANT_DENSITY, INC_IDEAL_GAS, INC_IDEAL_GAS_POLY)
FLUID_MODEL= STANDARD_AIR
%
% Ratio of specific heats (1.4 default and the value is hardcoded
%                          for the model STANDARD_AIR, compressible only)
GAMMA_VALUE= 1.4
%
% Specific gas constant (287.058 J/kg*K default and this value is hardcoded
%                        for the model STANDARD_AIR, compressible only)
GAS_CONSTANT= 287.058
%

% --------------------------- VISCOSITY MODEL ---------------------------------%
%
% Viscosity model (SUTHERLAND, CONSTANT_VISCOSITY, POLYNOMIAL_VISCOSITY).
VISCOSITY_MODEL= SUTHERLAND
%
%
% Sutherland Viscosity Ref (1.716E-5 default value for AIR SI)
MU_REF= 1.716E-5
%
% Sutherland Temperature Ref (273.15 K default value for AIR SI)
MU_T_REF= 273.15
%
% Sutherland constant (110.4 default value for AIR SI)
SUTHERLAND_CONSTANT= 110.4
%


% --------------------------- THERMAL CONDUCTIVITY MODEL ----------------------%
%
% Laminar Conductivity model (CONSTANT_CONDUCTIVITY, CONSTANT_PRANDTL, 
% POLYNOMIAL_CONDUCTIVITY).
CONDUCTIVITY_MODEL= CONSTANT_PRANDTL
%
%
% Laminar Prandtl number (0.72 (air), only for CONSTANT_PRANDTL)
PRANDTL_LAM= 0.72
%
% Definition of the turbulent thermal conductivity model for RANS
% (CONSTANT_PRANDTL_TURB by default, NONE).
TURBULENT_CONDUCTIVITY_MODEL= CONSTANT_PRANDTL_TURB
%
% Turbulent Prandtl number (0.9 (air) by default)
PRANDTL_TURB= 0.90


% ------------- COMMON PARAMETERS DEFINING THE NUMERICAL METHOD ---------------%
%
% Numerical method for spatial gradients (GREEN_GAUSS, WEIGHTED_LEAST_SQUARES)
NUM_METHOD_GRAD= GREEN_GAUSS
%
% CFL number (initial value for the adaptive CFL number)
CFL_NUMBER= 0.1
%
% Adaptive CFL number (NO, YES)
CFL_ADAPT= NO
%
% Parameters of the adaptive CFL number (factor down, factor up, CFL min value,
%                                        CFL max value )
CFL_ADAPT_PARAM= ( 0.8, 1.1, 0.3, 1.0)
%
% Maximum Delta Time in local time stepping simulations
MAX_DELTA_TIME= 1E6
%

MUSCL_FLOW= YES
MUSCL_TURB= NO
%
% Slope limiter (NONE, VENKATAKRISHNAN, VENKATAKRISHNAN_WANG,
%                BARTH_JESPERSEN, VAN_ALBADA_EDGE)
SLOPE_LIMITER_FLOW= VENKATAKRISHNAN
SLOPE_LIMITER_TURB= VENKATAKRISHNAN
%

VENKAT_LIMITER_COEFF= 0.3
%
%
% Freeze the value of the limiter after a number of iterations
LIMITER_ITER= 999999
%
%

% ------------------------ LINEAR SOLVER DEFINITION ---------------------------%
%
% Linear solver or smoother for implicit formulations:
% BCGSTAB, FGMRES, RESTARTED_FGMRES, CONJUGATE_GRADIENT (self-adjoint problems only), SMOOTHER.
LINEAR_SOLVER= FGMRES
%
%
% Preconditioner of the Krylov linear solver or type of smoother (ILU, LU_SGS, LINELET, JACOBI)
LINEAR_SOLVER_PREC= ILU
%

% Minimum error of the linear solver for implicit formulations
LINEAR_SOLVER_ERROR= 0.25
%
% Max number of iterations of the linear solver for the implicit formulation
LINEAR_SOLVER_ITER= 5
%


%USE_VECTORIZATION = YES % for only Roe
NEWTON_KRYLOV = NO
NEWTON_KRYLOV_IPARAM = (10, 3, 2) % n0, np, ft
NEWTON_KRYLOV_DPARAM = (1.0, 0.1, -6.0 1.e-5) % r0, tp, rt, e

% -------------------- FLOW NUMERICAL METHOD DEFINITION -----------------------%
%
% Convective numerical method (JST, LAX-FRIEDRICH, CUSP, ROE, AUSM, AUSMPLUSUP,
%                              AUSMPLUSUP2, HLLC, TURKEL_PREC, MSW, FDS, SLAU, SLAU2)
CONV_NUM_METHOD_FLOW= L2ROE
%
%
% Entropy fix coefficient (0.0 implies no entropy fixing, 1.0 implies scalar
%                          artificial dissipation)
ENTROPY_FIX_COEFF= 0.05
%
%
% Time discretization (RUNGE-KUTTA_EXPLICIT, EULER_IMPLICIT, EULER_EXPLICIT)
TIME_DISCRE_FLOW= EULER_IMPLICIT
TIME_DISCRE_TURB= EULER_IMPLICIT

% Convective numerical method (SCALAR_UPWIND)
CONV_NUM_METHOD_TURB= SCALAR_UPWIND

CFL_REDUCTION_TURB= 1.0

% ------------------------- SCREEN/HISTORY VOLUME OUTPUT --------------------------%
%
% Screen output fields (use 'SU2_CFD -d <config_file>' to view list of available fields)
SCREEN_OUTPUT= (INNER_ITER, WALL_TIME, RMS_DENSITY, RMS_MOMENTUM-X, RMS_MOMENTUM-Y, RMS_ENERGY,RMS_NU_TILDE,MIN_CFL, MAX_CFL, AOA, AVG_PRESS)
%
% History output groups (use 'SU2_CFD -d <config_file>' to view list of available fields)
HISTORY_OUTPUT= (ITER, RMS_RES)
%
% Volume output fields/groups (use 'SU2_CFD -d <config_file>' to view list of available fields)
VOLUME_OUTPUT= (COORDINATES, SOLUTION, PRIMITIVE)
%
% Writing frequency for screen output
SCREEN_WRT_FREQ_INNER= 1
%
SCREEN_WRT_FREQ_OUTER= 1
% 
SCREEN_WRT_FREQ_TIME= 1
%
% Writing frequency for history output
HISTORY_WRT_FREQ_INNER= 1
%
HISTORY_WRT_FREQ_OUTER= 1
% 
HISTORY_WRT_FREQ_TIME= 1
%
% Writing frequency for volume/surface output
OUTPUT_WRT_FREQ= 500
%
% ------------------------- INPUT/OUTPUT FILE INFORMATION --------------------------%
%
% Mesh input file
MESH_FILENAME= cone2daxi.su2
%
% Mesh input file format (SU2, CGNS)
MESH_FORMAT= SU2
%
% Mesh output file
MESH_OUT_FILENAME= mesh_out.su2
%
% Restart flow input file
SOLUTION_FILENAME= solution1st.dat
%

%
% Output tabular file format (TECPLOT, CSV)
TABULAR_FORMAT= TECPLOT
%
% Files to output 
% Possible formats : (TECPLOT, TECPLOT_BINARY, SURFACE_TECPLOT,
%  SURFACE_TECPLOT_BINARY, CSV, SURFACE_CSV, PARAVIEW, PARAVIEW_BINARY, SURFACE_PARAVIEW, 
%  SURFACE_PARAVIEW_BINARY, MESH, RESTART_BINARY, RESTART_ASCII, CGNS, STL)
% default : (RESTART, PARAVIEW, SURFACE_PARAVIEW)
%
% Files to output 
%OUTPUT_FILES= (RESTART, TECPLOT, SURFACE_TECPLOT)
OUTPUT_FILES= (RESTART, PARAVIEW, SURFACE_PARAVIEW)
%
% Output file convergence history (w/o extension)
CONV_FILENAME= history
%
% Output file with the forces breakdown
BREAKDOWN_FILENAME= forces_breakdown.dat
%
% Output file restart flow
RESTART_FILENAME= solution2nd.dat
%
%
% Output file flow (w/o extension) variables
VOLUME_FILENAME= flow
%
% Output file surface flow coefficient (w/o extension)
SURFACE_FILENAME= surface_flow
%
%
% Read binary restart files (YES, NO)
READ_BINARY_RESTART= YES
%

%==================== BOUNDARY CONDITIONS ==================================%
%MARKER_HEATFLUX= ( wall, 0.0)
MARKER_ISOTHERMAL = (wall, 300.0)
MARKER_FAR= ( farfield )
MARKER_SUPERSONIC_OUTLET= ( exit)
MARKER_SYM = (symmetry)
%==================== BOUNDARY CONDITIONS END ==============================%
%
MARKER_PLOTTING = ( wall)
MARKER_MONITORING = ( farfield, wall )
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%
%          			    E  N  D                                 %
% 													                        %
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

**Reply by WallyMaier** [reply]
Thanks.  I know you mentioned the adiabatic wall works well.  
I have run similar cases with no issues on the nose,  I am wondering if the isothermal wall and the L2ROE schemes don't play well together at higher Mach numbers.  Could you run a case with AUSM scheme?   It should help avoid any sort of carbuncle effects.  (Though the shock position looks fine.) 

Also what does your mesh look like at the nose?

**Reply by kohyun** [reply]
Thanks for your reply.

I will get back soon after AUSM run.

FYI, Following is my mesh around nose.
![image](https://user-images.githubusercontent.com/25341164/187229817-3482fbfa-9f35-4eb5-971e-df029be8b155.png)

**Comment by WallyMaier** [comment]
Interesting. The mesh looks to well-enough resolved at nose. There may be issues with symmetry plan/axis of rotation and heatflux.

**Reply by kohyun** [reply]
I have calculated using AUSMPLUSUP scheme.
Still, the wall temperature is not constant. (imposed 300K)
![image](https://user-images.githubusercontent.com/25341164/187323954-60b7dcc7-7f1f-4b65-896f-b3058b60c712.png)

following is a Mach contour.
![image](https://user-images.githubusercontent.com/25341164/187324120-38de2fef-c478-4530-aab1-c2b025d138c6.png)

**Comment by pcarruscag** [comment]
You are running this with CFL 0.1, has the solver converged?

**Reply by kohyun** [reply]
Thanks pcarruscag.

I have run the simulation with 1st order with following CFL adaptation.
<img width="727" alt="image" src="https://user-images.githubusercontent.com/25341164/187346246-14e59bad-0d14-4939-aea4-64a8b469efb9.png">

This is the residual plot of 1st order.
<img width="636" alt="image" src="https://user-images.githubusercontent.com/25341164/187346123-e658dac4-81af-4710-a411-6142e89ed453.png">

Used restart-run with 2nd order accuracy without CFL adaptation.

<img width="736" alt="image" src="https://user-images.githubusercontent.com/25341164/187346864-c8e6a6fa-3235-4224-923a-2fcf01cbe7f8.png">

You can see the residual plot(2nd order restart-run) here.
<img width="639" alt="image" src="https://user-images.githubusercontent.com/25341164/187346479-41c919f4-1adb-4e6a-8d5c-47f295ccb6cf.png">

If you want to check this case in your side, I can share the mesh and cfg files.

**Reply by pcarruscag** [reply]
That doesnt look good. Try something along the lines of the SWBLI case from the v&v page.

**Reply by pcarruscag** [reply]
From the Re number in your settings this does not seem to be a laminar case.

**Reply by kohyun** [reply]
> From the Re number in your settings this does not seem to be a laminar case.

Does Re affects the solution results when I used the following options?
================================================================
MACH_NUMBER= 6.0
INIT_OPTION= TD_CONDITIONS
FREESTREAM_OPTION= TEMPERATURE_FS
FREESTREAM_PRESSURE= 673.642
FREESTREAM_TEMPERATURE= 67.073
REF_DIMENSIONALIZATION= DIMENSIONAL
================================================================

When I check the running log file, the free-stream flow quantities (velocity, pressure, temperature etc) are correctly imposed.

Does Re number has any effects on the calculation in this case?

**Reply by kohyun** [reply]
> That doesnt look good. Try something along the lines of the SWBLI case from the v&v page.

I will check and report after trying that V&V cases.

**Reply by dylewiczk** [reply]
Hey @kohyun, I am running very similar geometry at very similar conditions (lower unit Reynolds number though) and I also have a problem with isothermal BC not being maintained along the wall.

Have you managed to troubleshoot this problem?

**Reply by pcarruscag** [reply]
The isothermal boundary is imposed weakly, that is, a heat flux is computed with the isothermal temperature and the temperature just inside the domain.
This implementation is conservative, imposing the temperature strongly at the wall node is not conservative (energy in and out of the domain will not be conserved).
Unfortunately, vertex-centered schemes are a bit weird.

**Reply by dylewiczk** [reply]
Makes sense, thank you!
